### PR TITLE
[AC-2647] Remove Flexible Collections MVP code

### PR DESCRIPTION
**Pull request by vincentsalucci**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/AC-2647

## 📔 Objective

> Remove dead code in the clients regarding the Flexible Collections MVP. 

* Update `organization.ts`
* Update AccessSelectorComponent and models
* Update CollectionView and CollectionAdminView
* Remove misc references in components --> `member-dialog` removed a large chunk of old permissions UI
* Delete AccessAll from request/response models --> There were a couple of potentially tricky changes to mapper functions that I'd like for reviewers to pay attention to

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: apps/web/src/app/admin-console/organizations/shared/components/access-selector/access-selector.models.ts [hunk 1]
```diff
@@ -140,8 +134,7 @@ export function mapGroupToAccessItemView(group: GroupView): AccessItemView {
     type: AccessItemType.Group,
     listName: group.name,
     labelName: group.name,
-    accessAllItems: group.accessAll,
-    readonly: group.accessAll,
+    readonly: false,
```
Comment (eliykat) [inline apps/web/src/app/admin-console/organizations/shared/components/access-selector/access-selector.models.ts:137]: I think you can just remove the `readonly` assignment here altogether. It's an optional property.

File: apps/web/src/app/admin-console/organizations/shared/components/access-selector/access-selector.models.ts [hunk 2]
```diff
@@ -155,7 +148,6 @@ export function mapUserToAccessItemView(user: OrganizationUserUserDetailsRespons
     listName: user.name?.length > 0 ? `${user.name} (${user.email})` : user.email,
     labelName: user.name ?? user.email,
     status: user.status,
-    accessAllItems: user.accessAll,
-    readonly: user.accessAll,
+    readonly: false,
```
Comment (eliykat) [inline apps/web/src/app/admin-console/organizations/shared/components/access-selector/access-selector.models.ts:151]: Same here

File: apps/web/src/app/vault/components/collection-dialog/collection-dialog.component.ts [hunk 3]
```diff
@@ -440,8 +440,7 @@ function mapGroupToAccessItemView(group: GroupView, collectionId: string): Acces
     type: AccessItemType.Group,
     listName: group.name,
     labelName: group.name,
-    accessAllItems: group.accessAll,
-    readonly: group.accessAll,
+    readonly: false,
```
Comment (eliykat) [inline apps/web/src/app/vault/components/collection-dialog/collection-dialog.component.ts:443]: And here. Although in this case, do we still need `readonlyPermission` in the next line? Will it ever be displayed?

File: apps/web/src/app/vault/components/collection-dialog/collection-dialog.component.ts [hunk 4]
```diff
@@ -467,8 +466,7 @@ function mapUserToAccessItemView(
     listName: user.name?.length > 0 ? `${user.name} (${user.email})` : user.email,
     labelName: user.name ?? user.email,
     status: user.status,
-    accessAllItems: user.accessAll,
-    readonly: user.accessAll,
+    readonly: false,
```
Comment (eliykat) [inline apps/web/src/app/vault/components/collection-dialog/collection-dialog.component.ts:469]: And here, and same question re: `readonlyPermission`.

**Review by eliykat — CHANGES_REQUESTED** [review]
I've reviewed commit-by-commit as recommended, mostly looks good, just a few items/questions to make sure we get it all.

**Review by audrey-jensen — APPROVED** [review]

File: apps/web/src/app/vault/components/collection-dialog/collection-dialog.component.ts (hunk 3 above)
Comment (vincentsalucci) [reply apps/web/src/app/vault/components/collection-dialog/collection-dialog.component.ts:443]: Yes! That property is used to display the assigned permission when the dialog is in a `disabled` (aka view only) state.

File: apps/web/src/app/vault/components/collection-dialog/collection-dialog.component.ts (hunk 4 above)
Comment (vincentsalucci) [reply apps/web/src/app/vault/components/collection-dialog/collection-dialog.component.ts:469]: Same answer!

File: libs/common/src/admin-console/models/domain/organization.ts [hunk 5]
```diff
@@ -220,7 +204,7 @@ export class Organization {
     restrictProviderAccessFlagEnabled: boolean,
   ) {
     // Before Flexible Collections, any admin or anyone with editAnyCollection permission could edit all ciphers
-    if (!this.flexibleCollections || !flexibleCollectionsV1Enabled || !this.flexibleCollections) {
+    if (!flexibleCollectionsV1Enabled) {
       return this.isAdmin || this.permissions.editAnyCollection;
     }
```
[the remark this answers is not in the document]
Comment (vincentsalucci) [reply libs/common/src/admin-console/models/domain/organization.ts:209]: Updated!

File: apps/web/src/app/admin-console/organizations/members/components/member-dialog/member-dialog.component.html [hunk 6]
```diff
@@ -49,14 +49,6 @@
               <bit-label>{{ "user" | i18n }}</bit-label>
               <bit-hint>{{ "userDesc" | i18n }}</bit-hint>
             </bit-radio-button>
-            <bit-radio-button
-              *ngIf="!organization.flexibleCollections"
-              id="userTypeManager"
-              [value]="organizationUserType.Manager"
-            >
-              <bit-label>{{ "manager" | i18n }}</bit-label>
-              <bit-hint>{{ "managerDesc" | i18n }}</bit-hint>
```
[the remark this answers is not in the document]
Comment (vincentsalucci) [reply apps/web/src/app/admin-console/organizations/members/components/member-dialog/member-dialog.component.html:58]: Updated!

File: apps/web/src/app/admin-console/organizations/shared/components/access-selector/access-selector.models.ts (hunk 1 above)
Comment (vincentsalucci) [reply apps/web/src/app/admin-console/organizations/shared/components/access-selector/access-selector.models.ts]: Removed

File: apps/web/src/app/admin-console/organizations/shared/components/access-selector/access-selector.models.ts (hunk 2 above)
Comment (vincentsalucci) [reply apps/web/src/app/admin-console/organizations/shared/components/access-selector/access-selector.models.ts]: Removed

**Review by eliykat — APPROVED** [review]
LGTM! I note that the deprecated custom permissions are still in `permissions.api.ts` but they're covered by a separate ticket, AC-1722, so I don't think we need to remove them here.

**Review by audrey-jensen — APPROVED** [review]

**Review by LRNcardozoWDF — APPROVED** [review]

**Review by shane-melton — APPROVED** [review]

**Pull request merged**